src/linter/index.tsx

```diff
@@ -0,0 +1,78 @@
+type Message = {
+  column: number | null;
+  description: string | string[];
+  file: string;
+  line: number | null;
+  message: string;
+  type: 'notice' | 'error' | 'warning';
+  uid: string;
+};
+
+export type ExternalMessage = Message & {
+  // These are some extra properties that we don't need to work with.
+  context: string[];
+  for_appversions: object;
+  id: string[];
+  tier: number;
+};
+
+type ExternalLinterResult = {
+  error: null | string;
+  full_report_url: string;
+  upload: string;
+  url: string;
+  validation: {
+    detected_type: string;
+    ending_tier: number;
+    errors: number;
+    message_tree: object;
+    messages: ExternalMessage[];
+    metadata: object;
+    notices: number;
+    success: boolean;
+    warnings: number;
+  };
+};
+
+const createInternalMessage = (message: ExternalMessage): Message => {
+  return {
+    column: message.column,
+    description: message.description,
```

Comment:
Related to my comment above, a trick could be to `[].concat(message.description)` to return an array in any case.

Reply:
Good idea, I like it

Reply:
TypeScript says no: `[].concat()` only accepts an array value (which MDN seems to confirm).

Reply:
mmh, MDN says `Arrays and/or values to concatenate into a new array`. I think TS is too picky here because this is one of the main purposes of `concat()`.

Reply:
Oh, I see, mdn does say non-array values are allowed. That's what I thought at first when I started to get lost in a rabbit hole of TypeScript issues. Oh, well, this reads perfectly fine:

```js
    description: Array.isArray(message.description)
      ? message.description
      : [message.description],
```